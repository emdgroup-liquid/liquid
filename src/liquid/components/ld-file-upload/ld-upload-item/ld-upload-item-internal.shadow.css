:host {
  align-items: flex-start;
  background-color: var(--ld-upload-item-background-color, var(--ld-col-wht));
  box-sizing: border-box;
  display: flex;
  flex-direction: column;
  gap: var(--ld-upload-item-gap, var(--ld-sp-8));
  list-style: none;
  margin: 0;
  min-height: var(--ld-upload-item-width, 4.875rem);
  max-width: var(--ld-upload-item-max-width, 100%);
  padding: var(
    --ld-upload-item-padding,
    var(--ld-sp-16) var(--ld-sp-8) var(--ld-sp-16) var(--ld-sp-16)
  );
  width: var(--ld-upload-item-width, 100%);
}

.ld-upload-item__container {
  display: flex;
  align-items: center;
  gap: var(--ld-upload-item-container-gap, var(--ld-sp-8));
  justify-items: start;
  width: 100%;
}

.ld-upload-item__icon {
  width: 2rem;
  height: auto;
  color: var(--ld-thm-primary);
  display: flex;
  justify-content: center;
}

::slotted(*) {
  width: 2rem;
  height: auto;
}

.ld-upload-item__file-details {
  align-items: flex-start;
  display: flex;
  flex: 1 1 0; /* required to stop flexbox from overflowing */
  flex-direction: column;
  margin-left: var(--ld-sp-8);
  min-width: 0; /* required to stop flexbox from overflowing */
}

.ld-upload-item__file-name,
.ld-upload-item__file-size {
  flex: 1;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  width: 100%;
}

.ld-upload-item__file-name {
  margin-bottom: var(--ld-sp-8);
}

.ld-upload-item__file-size {
  color: var(--ld-col-neutral-600);

  &::part(tag) {
    font-weight: normal;
  }
}

.ld-upload-item__buttons {
  display: flex;
  flex-direction: row;
  justify-self: end;
  justify-items: end;
}

.ld-upload-item__hide-on-sm {
  @media (width <= 32rem) {
    display: none;
  }
}

.ld-upload-item__progress {
  display: grid;
  width: 100%;
  justify-content: center;
}
