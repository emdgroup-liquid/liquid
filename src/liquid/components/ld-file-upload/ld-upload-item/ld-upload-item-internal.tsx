import {
  Component,
  Element,
  Host,
  h,
  Prop,
  Event,
  EventEmitter,
} from '@stencil/core'
import { getClassNames } from '../../../utils/getClassNames'
import { closest } from '../../../utils/closest'
import type { LdUploadItem } from '../ld-file-upload'

/**
 * @internal
 * @virtualProp ref - reference to component
 */
@Component({
  tag: 'ld-upload-item-internal',
  styleUrl: 'ld-upload-item-internal.shadow.css',
  shadow: true,
})
export class LdUploadItemInternal {
  @Element() el: HTMLLdUploadItemInternalElement

  /** Defines whether the user will be able to pause uploads. */
  @Prop() allowPause?: boolean

  /** Defines whether the total progress of all upoading files will be shown in the progress button */
  @Prop() showProgress?: boolean = false

  /** Tab index of the progress item. */
  @Prop() ldTabindex?: number

  /** State of the file. */
  @Prop() state?: LdUploadItem['state'] = 'pending'

  /** Name of the uploaded file. */
  @Prop() fileName: string

  /** Size of the uploaded file in bytes. */
  @Prop() fileSize: number

  /** Type of the uploaded file. */
  @Prop() fileType?: string

  /** Upload progress in percent. */
  @Prop() progress?: number = 0

  /** File of type File. */
  @Prop() file?: File

  /** List of all files currently in component */
  @Prop() uploadItems: LdUploadItem[] = []

  /** Label to be used for the tooltip of the cancel button. */
  @Prop() labelCancel: string

  /** Label to be used for the tooltip of the download button. */
  @Prop() labelDownload: string

  /** Label to be used for the tooltip of the remove button. */
  @Prop() labelRemove: string

  /** Label to be used for the tooltip of the retry button. */
  @Prop() labelRetry: string

  /** Label to be used for upload success message. */
  @Prop() labelUploadSuccessMsg: string

  /** Label to be used for upload cancelled message. */
  @Prop() labelUploadCancelledMsg: string

  /** Label to be used for upload error message. */
  @Prop() labelUploadErrorMsg: string

  /**
   * @internal
   * Emitted on pause button click.
   */
  @Event() lduploaditempause: EventEmitter<LdUploadItem>

  /**
   * @internal
   * Emitted on cancel button click.
   */
  @Event() lduploaditemcancel: EventEmitter<LdUploadItem>

  /**
   * @internal
   * Emitted on continue button click.
   */
  @Event() lduploaditemcontinue: EventEmitter<LdUploadItem>

  /**
   * @internal
   * Emitted on remove button click.
   */
  @Event() lduploaditemremove: EventEmitter<LdUploadItem>

  /**
   * @internal
   * Emitted on download button click.
   */
  @Event() lduploaditemdownload: EventEmitter<LdUploadItem>

  /**
   * @internal
   * Emitted on retry button click.
   */
  @Event() lduploaditemretry: EventEmitter<LdUploadItem>

  private removeClick = () => {
    this.lduploaditemremove.emit({
      state: this.state,
      fileName: this.fileName,
      fileSize: this.fileSize,
      fileType: this.fileType,
      progress: this.progress,
      file: this.file,
    })
  }

  private cancelClick = () => {
    this.lduploaditemcancel.emit({
      state: this.state,
      fileName: this.fileName,
      fileSize: this.fileSize,
      fileType: this.fileType,
      progress: this.progress,
      file: this.file,
    })
  }

  private downloadClick = () => {
    this.lduploaditemdownload.emit({
      state: this.state,
      fileName: this.fileName,
      fileSize: this.fileSize,
      fileType: this.fileType,
      progress: this.progress,
      file: this.file,
    })
  }

  private retryClick = () => {
    this.lduploaditemretry.emit({
      state: this.state,
      fileName: this.fileName,
      fileSize: this.fileSize,
      fileType: this.fileType,
      progress: this.progress,
      file: this.file,
    })
  }

  private bytesToSize = (bytes: number) => {
    const sizes = ['Bytes', 'KB', 'MB', 'GB', 'TB']

    let sizeIndex = 0

    while (bytes >= 1024 && sizeIndex < sizes.length - 1) {
      bytes /= 1024
      sizeIndex++
    }

    const roundedSize = bytes.toFixed(2)

    return `${roundedSize} ${sizes[sizeIndex]}`
  }

  componentWillLoad() {
    const customIcon = closest('ld-file-upload', this.el)?.querySelector(
      `[data-upload-icon='${this.fileType}']`
    )
    // If custom icon exists, clone it and insert it.
    if (customIcon) {
      const clonedIcon = customIcon.cloneNode(true) /* as HTMLElement */
      this.el.appendChild(clonedIcon)
    }
  }

  render() {
    const cl = getClassNames(['ld-upload-item'])

    const activeUploads = this.state === 'paused' || this.state === 'uploading'

    return (
      <Host class={cl}>
        <div class="ld-upload-item__container">
          <div class="ld-upload-item__icon">
            <slot name="icons">
              <ld-icon name="documents" size="lg" />
            </slot>
          </div>

          <div class="ld-upload-item__file-details">
            <ld-typo
              class="ld-upload-item__file-name"
              title={this.fileName}
              variant="label-s"
            >
              {this.fileName}
            </ld-typo>
            <ld-typo class="ld-upload-item__file-size" variant="label-s">
              {this.state === 'uploaded' || !this.showProgress
                ? this.bytesToSize(this.fileSize)
                : `${this.bytesToSize(
                    this.fileSize * (this.progress / 100)
                  )} / ${this.bytesToSize(this.fileSize)}`}
            </ld-typo>
          </div>
          <div class="ld-upload-item__buttons">
            {['pending', 'cancelled', 'uploaded', 'uploadFailed'].includes(
              this.state
            ) && (
              <ld-button
                class="ld-upload-item__remove-button"
                mode="ghost"
                size="sm"
                onClick={this.removeClick}
                slot="trigger"
              >
                <ld-icon
                  class="ld-upload-item__remove-icon"
                  name="cross"
                  size="sm"
                  aria-label="Text"
                />
                <div class="ld-upload-item__hide-on-sm">{this.labelRemove}</div>
              </ld-button>
            )}
            {['paused', 'uploading'].includes(this.state) && (
              <ld-button
                class="ld-upload-item__cancel-button"
                mode="ghost"
                size="sm"
                onClick={this.cancelClick}
                slot="trigger"
              >
                <ld-icon
                  class="ld-upload-item__cancel-icon"
                  name="cross"
                  size="sm"
                  aria-label="Text"
                />
                <div class="ld-upload-item__hide-on-sm">{this.labelCancel}</div>
              </ld-button>
            )}
            {this.state === 'uploaded' && (
              <ld-button
                class="ld-upload-item__download-button"
                mode="ghost"
                size="sm"
                onClick={this.downloadClick}
                slot="trigger"
              >
                <ld-icon
                  class="ld-upload-item__download-icon"
                  name="download"
                  size="sm"
                  aria-label="Text"
                />
                <div class="ld-upload-item__hide-on-sm">
                  {this.labelDownload}
                </div>
              </ld-button>
            )}
            {this.state === 'uploadFailed' &&
              this.uploadItems.filter((item) => item.state === 'uploading')
                .length === 0 && (
                <ld-button
                  class="ld-upload-item__retry-button"
                  mode="ghost"
                  size="sm"
                  onClick={this.retryClick}
                  slot="trigger"
                >
                  <ld-icon
                    class="ld-upload-item__retry-icon"
                    name="refresh"
                    size="sm"
                    aria-label="Text"
                  />
                  <div class="ld-upload-item__hide-on-sm">
                    {this.labelRetry}
                  </div>
                </ld-button>
              )}
          </div>
        </div>
        {['pending', 'uploading'].includes(this.state) && (
          <ld-sr-only id="progress-label">Progress</ld-sr-only>
        )}
        {activeUploads && this.progress !== 0 ? (
          <ld-progress
            class="ld-upload-item__progress"
            pending={this.state === 'paused'}
            aria-labeledby="progress-label"
            aria-valuenow={this.progress}
          />
        ) : (
          activeUploads &&
          this.progress === 0 && (
            <ld-progress
              class="ld-upload-item__progress"
              pending
              aria-labeledby="progress-label"
              aria-valuetext="indeterminate"
            ></ld-progress>
          )
        )}
        {this.state === 'cancelled' && (
          <ld-input-message mode="info">
            {this.labelUploadCancelledMsg}
          </ld-input-message>
        )}
        {this.state === 'uploaded' && (
          <ld-input-message mode="valid">
            {this.labelUploadSuccessMsg}
          </ld-input-message>
        )}
        {this.state === 'uploadFailed' && (
          <ld-input-message>{this.labelUploadErrorMsg}</ld-input-message>
        )}
        <slot />
      </Host>
    )
  }
}
