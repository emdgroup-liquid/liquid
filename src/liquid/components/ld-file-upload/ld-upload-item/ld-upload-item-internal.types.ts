export interface LdUploadItemConfig {
  pdf?: string
  zip?: string
  jpeg?: string
  txt?: string
  png?: string
  rtf?: string
}
