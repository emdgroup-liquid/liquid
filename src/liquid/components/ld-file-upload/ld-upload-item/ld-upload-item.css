:host {
  background: var(--ld-col-wht);
  box-sizing: border-box;
  list-style: none;
  margin: 0;
  padding: var(--ld-sp-24);
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  max-width: 100%;
  gap: var(--ld-sp-8);
}

.ld-upload-item__container {
  display: flex;
  align-items: center;
  gap: var(--ld-sp-24);
  justify-items: start;
  width: 100%;
}

.ld-upload-item__icon {
  width: 2rem;
  height: auto;
  /* display: flex;
  justify-content: center; */
}

.ld-upload-item__custom-icon {
  width: 2rem;
  height: auto;
}

::slotted(*) {
  width: 2rem;
  height: auto;
}

.ld-upload-item__file-details {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  flex-grow: 1;
  /* width: calc(100% - 9rem); */
  width: calc(100% - 18rem);
}

.ld-upload-item__file-name {
  flex: 1;
  width: 100%;

  &::part(tag) {
    overflow: hidden;
    text-overflow: ellipsis;
  }
}

.ld-upload-item__file-size {
  flex: 1;
  width: 100%;

  &::part(tag) {
    overflow: hidden;
    text-overflow: ellipsis;
  }
}

.ld-upload-item__buttons {
  display: flex;
  flex-direction: row;
  justify-self: end;
  justify-items: end;
}

.ld-upload-item__pause-icon {
  color: var(--ld-col-neutral-900);
}

.ld-upload-item__continue-icon {
  color: var(--ld-col-neutral-900);
}

.ld-upload-item__remove-icon {
  color: var(--ld-col-neutral-900);
}

.ld-upload-item__download-icon {
  color: var(--ld-col-neutral-900);
}

.ld-upload-item__retry-icon {
  color: var(--ld-col-neutral-900);
}

.ld-upload-item__delete-icon {
  color: var(--ld-col-rr-600);
}

.ld-upload-item__progress {
  display: grid;
  width: 100%;
  justify-content: center;
}
