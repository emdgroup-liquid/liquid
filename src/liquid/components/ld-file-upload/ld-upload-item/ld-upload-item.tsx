import {
  Component,
  Element,
  Host,
  h,
  Prop,
  Event,
  EventEmitter,
} from '@stencil/core'
import { getClassNames } from '../../../utils/getClassNames'
import { closest } from '../../../utils/closest'

/**
 * @virtualProp ref - reference to component
 */
@Component({
  tag: 'ld-upload-item',
  styleUrl: 'ld-upload-item.css',
  shadow: true,
})
export class LdUploadItem {
  @Element() el: HTMLLdUploadItemElement

  /** allowPause defines whether the user will be able to pause uploads. */
  @Prop() allowPause?: boolean

  /** showTotalProgress defines whether the total progress of all upoading files will be shown in the progress button */
  @Prop() showProgress?: boolean = false

  /** State of the file. */
  @Prop() state?:
    | 'pending'
    | 'paused'
    | 'cancelled'
    | 'uploading'
    | 'uploaded'
    | 'upload failed' = 'pending'

  /** Name of the uploaded file. */
  @Prop() fileName: string

  /** Size of the uploaded file in bytes. */
  @Prop() fileSize: number

  /** Type of the uploaded file. */
  @Prop() fileType?: string

  /** Tab index of the progress item. */
  @Prop() ldTabindex?: number

  /**
   * Size of the menu item.
   * @internal
   */
  @Prop() size?: 'sm' | 'lg'

  /** Upload progress in percent. */
  @Prop() progress?: number = 0

  /**
   * @internal
   * Emitted on pause button click.
   */
  @Event() lduploaditempause: EventEmitter

  /**
   * @internal
   * Emitted on continue button click.
   */
  @Event() lduploaditemcontinue: EventEmitter

  /**
   * @internal
   * Emitted on stop button click.
   */
  @Event() lduploaditemremove: EventEmitter

  /**
   * @internal
   * Emitted on download button click.
   */
  @Event() lduploaditemdownload: EventEmitter

  /**
   * @internal
   * Emitted on retry button click.
   */
  @Event() lduploaditemretry: EventEmitter

  /**
   * @internal
   * Emitted on delete button click.
   */
  @Event() lduploaditemdelete: EventEmitter

  private pauseClick = () => {
    this.lduploaditempause.emit({
      state: this.state,
      fileName: this.fileName,
      fileSize: this.fileSize,
      fileType: this.fileType,
      progress: this.progress,
    })
  }

  private continueUploadClick = () => {
    this.lduploaditemcontinue.emit({
      state: this.state,
      fileName: this.fileName,
      fileSize: this.fileSize,
      fileType: this.fileType,
      progress: this.progress,
    })
  }

  private removeClick = () => {
    this.lduploaditemremove.emit({
      state: this.state,
      fileName: this.fileName,
      fileSize: this.fileSize,
      fileType: this.fileType,
      progress: this.progress,
    })
  }

  private downloadClick = () => {
    this.lduploaditemdownload.emit({
      state: this.state,
      fileName: this.fileName,
      fileSize: this.fileSize,
      fileType: this.fileType,
      progress: this.progress,
    })
  }

  private retryClick = () => {
    this.lduploaditemretry.emit({
      state: this.state,
      fileName: this.fileName,
      fileSize: this.fileSize,
      fileType: this.fileType,
      progress: this.progress,
    })
  }

  private deleteClick = () => {
    this.lduploaditemdelete.emit({
      state: this.state,
      fileName: this.fileName,
      fileSize: this.fileSize,
      fileType: this.fileType,
      progress: this.progress,
    })
  }

  private bytesToSize = (bytes: number) => {
    const sizes = ['Bytes', 'KB', 'MB', 'GB', 'TB']

    let sizeIndex = 0

    while (bytes >= 1024 && sizeIndex < sizes.length - 1) {
      bytes /= 1024
      sizeIndex++
    }

    const roundedSize = Number(bytes.toFixed(2))

    return roundedSize + ' ' + sizes[sizeIndex]
  }

  componentWillLoad() {
    const customIcon = closest('ld-file-upload', this.el)?.querySelector(
      `[data-upload-icon='${this.fileType}']`
    )
    // TODO: If custom icon exists, clone it and insert it.
    if (customIcon) {
      // this.customIcon = customIcon.cloneNode(true)
      // this.customIcon.className = 'ld-upload-item__icon'
      // console.log(this.customIcon)
      const clonedIcon = customIcon.cloneNode(true) /* as HTMLElement */
      /* clonedIcon.style.width = '2rem'
      clonedIcon.style.height = 'auto' */
      // this.customIcon = <div class="ld-upload-item__icon">{clonedIcon}</div>
      this.el.appendChild(clonedIcon)
    } /* else {
      // this.customIcon = (
      //   <ld-icon
      //     class="ld-upload-item__icon"
      //     name="documents"
      //     size="lg"
      //   ></ld-icon>
      // )
      this.el.appendChild(
        <ld-icon slot="icons" name="documents" size="lg"></ld-icon>
      )
    } */

    /* if (customIcon) {
      this.customIcon = customIcon.cloneNode(true) as HTMLElement
    } */
  }

  render() {
    const cl = getClassNames(['ld-upload-item'])

    const activeUploads =
      this.state == 'pending' ||
      this.state == 'paused' ||
      this.state == 'uploading'

    return (
      <Host class={cl}>
        <div class="ld-upload-item__container">
          {/* überprüfen, ob es sich um einen Pfad oder einen Icon namen handelt und entsprechend img oder ld-icon nutzen */}
          {/* {this.availableIcons.includes(this.setIcon()) ? (
            <ld-icon
              class="ld-upload-item__icon"
              name={this.setIcon()}
              size="lg"
            ></ld-icon>
          ) : (
            <img class="ld-upload-item__icon" src={this.setIcon()}></img>
          )} */}
          {/* {this.customIcon} */}
          {/* {this.getIcon()} */}
          {/* <div class="ld-upload-item__icon">
            {this.customIcon ? (
              this.customIcon
            ) : (
              <ld-icon name="documents" size="lg"></ld-icon>
            )}
          </div> */}
          {
            <div class="ld-upload-item__icon">
              <slot name="icons">
                <pre>
                  <ld-icon name="documents" size="lg"></ld-icon>
                </pre>
              </slot>
            </div>
          }

          <div class="ld-upload-item__file-details">
            <ld-typo class="ld-upload-item__file-name" variant="h5">
              {this.fileName}
            </ld-typo>
            {/* <b class="ld-upload-item__file-name">{this.fileName}</b> */}
            {this.state == 'uploaded' || !this.showProgress ? (
              <ld-typo class="ld-upload-item__file-size">
                {this.bytesToSize(this.fileSize)}
              </ld-typo>
            ) : (
              <ld-typo class="ld-upload-item__file-size">
                {this.bytesToSize(this.fileSize * (this.progress / 100))} /{' '}
                {this.bytesToSize(this.fileSize)}
              </ld-typo>
            )}
          </div>
          <div class="ld-upload-item__buttons">
            {this.state == 'uploading' && this.allowPause ? (
              <ld-tooltip arrow position="left middle" size="sm">
                <ld-button
                  mode="ghost"
                  size="sm"
                  onClick={this.pauseClick}
                  slot="trigger"
                >
                  <ld-icon
                    class="ld-upload-item__pause-icon"
                    name="pause"
                    size="sm"
                    aria-label="Text"
                  ></ld-icon>
                </ld-button>
                <ld-typo>Pause</ld-typo>
              </ld-tooltip>
            ) : undefined}
            {this.state == 'paused' && this.allowPause ? (
              <ld-tooltip arrow position="left middle" size="sm">
                <ld-button
                  mode="ghost"
                  size="sm"
                  onClick={this.continueUploadClick}
                  slot="trigger"
                >
                  <ld-icon
                    class="ld-upload-item__continue-icon"
                    name="upload"
                    size="sm"
                    aria-label="Text"
                  ></ld-icon>
                </ld-button>
                <ld-typo>Continue upload</ld-typo>
              </ld-tooltip>
            ) : undefined}
            {this.state == 'pending' ||
            this.state == 'paused' ||
            this.state == 'uploading' ? (
              <ld-tooltip arrow position="left middle" size="sm">
                <ld-button
                  mode="ghost"
                  size="sm"
                  onClick={this.removeClick}
                  slot="trigger"
                >
                  <ld-icon
                    class="ld-upload-item__remove-icon"
                    name="cross"
                    size="sm"
                    aria-label="Text"
                  ></ld-icon>
                </ld-button>
                <ld-typo>Remove</ld-typo>
              </ld-tooltip>
            ) : undefined}
            {this.state == 'uploaded' ? (
              <ld-tooltip arrow position="left middle" size="sm">
                <ld-button
                  mode="ghost"
                  size="sm"
                  onClick={this.downloadClick}
                  slot="trigger"
                >
                  <ld-icon
                    class="ld-upload-item__download-icon"
                    name="download"
                    size="sm"
                    aria-label="Text"
                  ></ld-icon>
                </ld-button>
                <ld-typo>Download</ld-typo>
              </ld-tooltip>
            ) : undefined}
            {this.state == 'upload failed' ? (
              <ld-tooltip arrow position="left middle" size="sm">
                <ld-button
                  mode="ghost"
                  size="sm"
                  onClick={this.retryClick}
                  slot="trigger"
                >
                  <ld-icon
                    class="ld-upload-item__retry-icon"
                    name="refresh"
                    size="sm"
                    aria-label="Text"
                  ></ld-icon>
                </ld-button>
                <ld-typo>Retry</ld-typo>
              </ld-tooltip>
            ) : undefined}
            {this.state == 'cancelled' ||
            this.state == 'uploaded' ||
            this.state == 'upload failed' ? (
              <ld-tooltip arrow position="left middle" size="sm">
                <ld-button
                  mode="ghost"
                  size="sm"
                  onClick={this.deleteClick}
                  slot="trigger"
                >
                  <ld-icon
                    class="ld-upload-item__delete-icon"
                    name="bin"
                    size="sm"
                    aria-label="Text"
                  ></ld-icon>
                </ld-button>
                <ld-typo>Delete</ld-typo>
              </ld-tooltip>
            ) : undefined}
          </div>
        </div>
        {this.state == 'pending' || this.state == 'uploading' ? (
          <ld-sr-only id="progress-label">Progress</ld-sr-only>
        ) : undefined}
        {activeUploads && this.showProgress ? (
          <ld-progress
            class="ld-upload-item__progress"
            pending={this.state == 'paused'}
            aria-labeledby="progress-label"
            aria-valuenow={this.progress}
          ></ld-progress>
        ) : this.state == 'uploading' && !this.showProgress ? (
          <div class="ld-upload-item__progress">
            <ld-loading class="ld-upload-item__loading"></ld-loading>
          </div>
        ) : undefined}
        {this.state == 'cancelled' ? (
          <ld-input-message mode="info">
            Upload of this file has been cancelled
          </ld-input-message>
        ) : undefined}
        {this.state == 'uploaded' ? (
          <ld-input-message mode="valid">
            Upload was successful!
          </ld-input-message>
        ) : undefined}
        {this.state == 'upload failed' ? (
          <ld-input-message>Error! Upload was unsuccessful</ld-input-message>
        ) : undefined}
        <slot></slot>
      </Host>
    )
  }
}
