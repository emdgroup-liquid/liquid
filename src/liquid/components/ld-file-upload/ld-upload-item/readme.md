---
eleventyNavigation:
  key: Upload Item
  parent: File Upload
layout: layout.njk
title: Upload Item
permalink: components/ld-file-upload/ld-upload-item/
---

# ld-upload-item

The `ld-upload-item` component is a subcomponent for `ld-file-upload` / `ld-upload-progress` and is meant to be used in the slot of the [`ld-upload-progress`](../ld-upload-progress) component.

## Examples

### Default / Pending

{% example '{ "opened": false }' %}
<ld-upload-item file-name='Liquid' file-size='1.28'></ld-upload-item>

<!-- React component -->

<!-- CSS component -->

{% endexample %}

#### Show progress

{% example '{ "opened": false }' %}
<ld-upload-item show-progress file-name='Liquid' file-size='1.28'></ld-upload-item>

<!-- React component -->

<!-- CSS component -->

{% endexample %}

### Paused

{% example '{ "opened": false }' %}
<ld-upload-item allow-pause state='paused' file-name='Liquid' file-size='1.28' progress='25'></ld-upload-item>

<!-- React component -->

<!-- CSS component -->

{% endexample %}

#### Show progress

{% example '{ "opened": false }' %}
<ld-upload-item show-progress state='paused' file-name='Liquid' file-size='1.28' progress='25'></ld-upload-item>

<!-- React component -->

<!-- CSS component -->

{% endexample %}

### Cancelled

{% example '{ "opened": false }' %}
<ld-upload-item state='cancelled' file-name='Liquid' file-size='1.28' progress='25'></ld-upload-item>

<!-- React component -->

<!-- CSS component -->

{% endexample %}

### Uploading

{% example '{ "opened": false }' %}
<ld-upload-item state='uploading' file-name='Liquid' file-size='1.28' progress='25'></ld-upload-item>

<!-- React component -->

<!-- CSS component -->

{% endexample %}

#### Allow pause

{% example '{ "opened": false }' %}
<ld-upload-item allow-pause state='uploading' file-name='Liquid' file-size='1.28' progress='25'></ld-upload-item>

<!-- React component -->

<!-- CSS component -->

{% endexample %}

### Uploading interactive

{% example '{ "opened": false }' %}
<ld-upload-item show-progress state='uploading' file-name='Liquid' file-size='1.28' progress='25'></ld-upload-item>
<ld-slider value="25" max="100" width="14rem"></ld-slider>

<script>
  void function() {
    const slider = document.currentScript.previousElementSibling
    const progress = slider.previousElementSibling
    slider.addEventListener('ldchange', ev => {
      const val = ev.detail[0]
      progress.progress = val
    })
  }()
</script>

<!-- React component -->

<!-- CSS component -->

{% endexample %}

### Uploaded

{% example '{ "opened": false }' %}
<ld-upload-item state='uploaded' file-name='Liquid' file-size='1.28'></ld-upload-item>

<!-- React component -->

<!-- CSS component -->

{% endexample %}

### Upload failed

{% example '{ "opened": false }' %}
<ld-upload-item state='upload failed' file-name='Liquid' file-size='1.28'></ld-upload-item>

<!-- React component -->

<!-- CSS component -->

{% endexample %}

<!-- Auto Generated Below -->


## Properties

| Property       | Attribute       | Description                                                                                                     | Type                                                                                   | Default     |
| -------------- | --------------- | --------------------------------------------------------------------------------------------------------------- | -------------------------------------------------------------------------------------- | ----------- |
| `allowPause`   | `allow-pause`   | allowPause defines whether the user will be able to pause uploads.                                              | `boolean`                                                                              | `undefined` |
| `fileName`     | `file-name`     | Name of the uploaded file.                                                                                      | `string`                                                                               | `undefined` |
| `fileSize`     | `file-size`     | Size of the uploaded file in bytes.                                                                             | `number`                                                                               | `undefined` |
| `fileType`     | `file-type`     | Type of the uploaded file.                                                                                      | `string`                                                                               | `undefined` |
| `ldTabindex`   | `ld-tabindex`   | Tab index of the progress item.                                                                                 | `number`                                                                               | `undefined` |
| `progress`     | `progress`      | Upload progress in percent.                                                                                     | `number`                                                                               | `0`         |
| `ref`          | `ref`           | reference to component                                                                                          | `any`                                                                                  | `undefined` |
| `showProgress` | `show-progress` | showTotalProgress defines whether the total progress of all upoading files will be shown in the progress button | `boolean`                                                                              | `false`     |
| `state`        | `state`         | State of the file.                                                                                              | `"cancelled" \| "paused" \| "pending" \| "upload failed" \| "uploaded" \| "uploading"` | `'pending'` |


## Dependencies

### Used by

 - [ld-upload-progress](../ld-upload-progress)

### Depends on

- [ld-icon](../../ld-icon)
- [ld-typo](../../ld-typo)
- [ld-tooltip](../../ld-tooltip)
- [ld-button](../../ld-button)
- [ld-sr-only](../../ld-sr-only)
- [ld-progress](../../ld-progress)
- [ld-loading](../../ld-loading)
- [ld-input-message](../../ld-input-message)

### Graph
```mermaid
graph TD;
  ld-upload-item --> ld-icon
  ld-upload-item --> ld-typo
  ld-upload-item --> ld-tooltip
  ld-upload-item --> ld-button
  ld-upload-item --> ld-sr-only
  ld-upload-item --> ld-progress
  ld-upload-item --> ld-loading
  ld-upload-item --> ld-input-message
  ld-tooltip --> ld-sr-only
  ld-tooltip --> ld-tooltip-popper
  ld-input-message --> ld-icon
  ld-upload-progress --> ld-upload-item
  style ld-upload-item fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
