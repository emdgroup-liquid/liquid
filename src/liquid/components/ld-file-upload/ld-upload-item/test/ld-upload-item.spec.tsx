import { newSpecPage } from '@stencil/core/testing'
import { LdUploadItem } from '../ld-upload-item'

describe('ld-menuitem', () => {
  it('renders default', async () => {
    const page = await newSpecPage({
      components: [LdUploadItem],
      html: `<ld-upload-item file-name='Liquid' file-size='1.28'></ld-upload-item>`,
    })
    expect(page.root).toMatchSnapshot()
  })
})
