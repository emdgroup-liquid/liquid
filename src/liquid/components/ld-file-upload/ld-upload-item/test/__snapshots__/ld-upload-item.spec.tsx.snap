// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ld-menuitem renders default 1`] = `
<ld-upload-item class="ld-upload-item" file-name="Liquid" file-size="1.28">
  <mock:shadow-root>
    <ld-card class="ld-upload-item__card">
      <div class="ld-upload-item__container">
        <ld-icon class="ld-upload-item__icon" name="placeholder"></ld-icon>
        <div class="ld-upload-item__file-details">
          <ld-typo variant="h5">
            Liquid
          </ld-typo>
          <ld-typo>
            1.28 Bytes
          </ld-typo>
        </div>
        <div class="ld-upload-item__buttons">
          <ld-button mode="ghost">
            <ld-icon aria-label="Text" name="cross"></ld-icon>
          </ld-button>
        </div>
      </div>
      <ld-sr-only id="progress-label">
        Progress
      </ld-sr-only>
      <ld-progress aria-labeledby="progress-label" aria-valuenow="0" class="ld-upload-item__progress"></ld-progress>
      <slot></slot>
    </ld-card>
  </mock:shadow-root>
</ld-upload-item>
`;
